### PR TITLE
mgr/cephadm: fix orch rm and upgrade

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/pybind/mgr/orchestrator/_interface.py [hunk 1]
```diff
@@ -1667,14 +1667,20 @@ class RGWSpec(ServiceSpec):
...
+                 rgw_zone=None,  # type: Optional[str]
+                 service_id=None, # type: Optional[str]
                  placement=None,
                  service_type='rgw',
                  rgw_frontend_port=None,  # type: Optional[int]
                  ):
         assert service_type == 'rgw'
-        super(RGWSpec, self).__init__('rgw', service_id=rgw_realm+'.'+rgw_zone, placement=placement)
+        if service_id:
+            (rgw_realm, rgw_zone) = service_id.split('.', 1)
```
Comment (jschmid1) [inline src/pybind/mgr/orchestrator/_interface.py]: Not sure what this is doing?

Comment (liewegas) [reply src/pybind/mgr/orchestrator/_interface.py]: it allows you to pass service_id='realm.zone' and the ctor will still populate realm_id and zone_id.

**Comment by liewegas** [comment]
green: http://pulpito.ceph.com/sage-2020-03-06_07:54:49-rados:cephadm-wip-sage3-testing-2020-03-05-2154-distro-basic-smithi/

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

File: src/pybind/mgr/orchestrator/_interface.py (hunk 1 above)
Comment (jschmid1) [reply src/pybind/mgr/orchestrator/_interface.py]: oh, right the super() was moved below that. marking as resovled then

**Comment by jschmid1** [comment]
lgtm

**Pull request merged**